pkg/api/validate/dynamic/quota.go

```diff
@@ -133,16 +83,18 @@ func (dv *dynamic) ValidateQuota(ctx context.Context, oc *api.OpenShiftCluster)
 	}
 
 	requiredResources := map[string]int{}
-	err := addRequiredResources(requiredResources, oc.Properties.MasterProfile.VMSize, 3)
-	if err != nil {
-		return err
+	vm, ok := vmFromTypeName(oc.Properties.MasterProfile.VMSize)
+	if !ok {
+		return fmt.Errorf("unexpected node VMSize %s", oc.Properties.MasterProfile.VMSize)
```

Comment:
Can we return the specific profile that is unexpected?
```suggestion
		return fmt.Errorf("unexpected MasterProfile VMSize %s", oc.Properties.MasterProfile.VMSize)
```